jscomp/bsb/bsb_helper_dep_graph.mli

```diff
@@ -0,0 +1,32 @@
+(* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
+
+val sort_files_by_dependencies : domain:String_set.t -> String_set.t String_map.t -> string Queue.t
+
+val simple_collect_from_main :
+           ?alias_map:String_set.elt String_hashtbl.t ->
```

Comment:
Mmmh those are similar to what exist in `ast_extract`.

`sort_files_by_dependencies` was copied over and changed to remove the dependency on `bs_exception` which in turn depends on `compiler-libs`. I'm not sure how to make it better apart from refactoring bs_exception to not depend on compiler-libs.

`simple_collect_from_main` assumes that you already have the `ast_table` and doesn't allocate records like `Ast_extract.collection_from_main`. It's the same core logic though.